src/value.rs

```diff
@@ -0,0 +1,181 @@
+        };
+
+        let count = types::read_int_length(v.as_slice_mut()).map_err(DeserializationError::new)?;
+
+        let raw_iter = FixedLengthBytesSequenceIterator::new(count, v);
+
+        let list = PyList::empty(py);
+        for raw_opt in raw_iter {
+            let raw = raw_opt.map_err(DeserializationError::new)?;
+            let item = T::deserialize_py(elem_typ, raw, py)?;
+            list.append(item).map_err(DeserializationError::new)?;
+        }
+
+        Ok(PyDeserializedValue::new(list.into_any()))
+    }
+}
+
+impl<'frame, 'metadata, 'py, T> PyDeserializeValue<'frame, 'metadata, 'py> for Option<T>
+where
+    T: PyDeserializeValue<'frame, 'metadata, 'py>,
+{
+    fn deserialize_py(
+        typ: &'metadata ColumnType<'metadata>,
+        v: Option<FrameSlice<'frame>>,
+        py: Python<'py>,
+    ) -> Result<PyDeserializedValue<'py>, DeserializationError> {
+        match v {
+            Some(v) => T::deserialize_py(typ, Some(v), py),
+            None => Ok(PyDeserializedValue::new(py.None().bind(py).to_owned())),
+        }
+    }
+}
+
+pub struct PyDeserializedValue<'py> {
+    pub value: Bound<'py, PyAny>,
+}
+
+impl<'py> PyDeserializedValue<'py> {
+    fn new(value: Bound<'py, PyAny>) -> Self {
+        Self { value }
+    }
+}
+impl<'py> IntoPyObject<'py> for PyDeserializedValue<'py> {
+    type Target = PyAny;
+    type Output = Bound<'py, PyAny>;
+    type Error = Infallible;
+    fn into_pyobject(self, _: Python<'py>) -> Result<Self::Output, Self::Error> {
+        Ok(self.value)
+    }
+}
+
+impl<'frame, 'metadata, 'py> PyDeserializeValue<'frame, 'metadata, 'py>
+    for PyDeserializedValue<'py>
+{
+    fn deserialize_py(
+        typ: &'metadata ColumnType<'metadata>,
+        v: Option<FrameSlice<'frame>>,
+        py: Python<'py>,
+    ) -> Result<Self, DeserializationError> {
+        let cql = deser_cql_py_value(py, typ, v)?;
+
+        Ok(PyDeserializedValue { value: cql })
+    }
+}
+
+fn deser_cql_py_value<'py, 'metadata, 'frame>(
+    py: Python<'py>,
+    typ: &'metadata ColumnType<'metadata>,
+    val: Option<FrameSlice<'frame>>,
+) -> Result<Bound<'py, PyAny>, DeserializationError> {
+    match typ {
+        ColumnType::Native(native_type) => match native_type {
+            NativeType::Int => {
+                let v = i32::deserialize(typ, val)?;
+                Ok(PyInt::new(py, v).into_any())
+            }
+            NativeType::BigInt => {
+                let v = i64::deserialize(typ, val)?;
+                Ok(PyInt::new(py, v).into_any())
+            }
+            NativeType::Text => {
+                let v = <&str as DeserializeValue<'frame, 'metadata>>::deserialize(typ, val)?;
+                Ok(PyString::new(py, v).into_any())
+            }
+            NativeType::Boolean => {
+                let v = bool::deserialize(typ, val)?;
+                Ok(PyBool::new(py, v).to_owned().into_any())
+            }
```

Comment:
:question: What does this `to_owned()` do here? Why is this needed specifically for boolean, while not needed for other primitive types?

Reply:
This is OK IMO then.